addons/web/static/src/core/browser/router.js

```diff
@@ -36,53 +36,49 @@ function parseString(str) {
 /**
  * For each push request (replaceState or pushState), filterout keys that have been locked before
  * overrides locked keys that are explicitly re-locked or unlocked
- * registers keys in "hash" in "lockedKeys" according to the "lock" Boolean
+ * registers keys in "search" in "lockedKeys" according to the "lock" Boolean
  *
- * @param {Set<string>} lockedKeys A set containing all keys that were locked
- * @param {Query} hash An Object representing the pushed url hash
- * @param {Query} currentHash The current hash compare against
- * @param  {Object} [options={}] Whether to lock all hash keys in "hash" to prevent them from being changed afterwards
- * @param  {Boolean} [options.lock] Whether to lock all hash keys in "hash" to prevent them from being changed afterwards
- * @return {Query} The resulting "hash" where previous locking has been applied
+ * @param {Query} search An Object representing the pushed url search
+ * @param {Query} currentSearch The current search compare against
+ * @return {Query} The resulting "search" where previous locking has been applied
  */
-function applyLocking(lockedKeys, hash, currentHash, options = {}) {
-    const newHash = {};
-    for (const key in hash) {
-        if ("lock" in options) {
-            options.lock ? lockedKeys.add(key) : lockedKeys.delete(key);
-        } else if (lockedKeys.has(key)) {
-            // forbid implicit override of key
-            continue;
+function applyLocking(search, currentSearch) {
+    const newSearch = Object.assign({}, search);
+    for (const key in currentSearch) {
+        if ([..._lockedKeys].includes(key) && !(key in newSearch)) {
+            newSearch[key] = currentSearch[key];
         }
-        newHash[key] = hash[key];
     }
-    for (const key in currentHash) {
-        if (lockedKeys.has(key) && !(key in newHash)) {
-            newHash[key] = currentHash[key];
-        }
-    }
-    return newHash;
+    return newSearch;
 }
 
-function computeNewRoute(hash, replace, currentRoute) {
+function computeNewRoute(search, replace, currentRoute) {
     if (!replace) {
-        hash = Object.assign({}, currentRoute.hash, hash);
+        search = Object.assign({}, currentRoute, search);
     }
-    hash = sanitizeHash(hash);
-    if (!shallowEqual(currentRoute.hash, hash)) {
-        return Object.assign({}, currentRoute, { hash });
+    search = sanitizeSearch(search);
+    if (!shallowEqual(currentRoute, search)) {
+        return search;
     }
     return false;
 }
 
-function sanitizeHash(hash) {
+function sanitize(obj, valueToRemove) {
     return Object.fromEntries(
-        Object.entries(hash)
-            .filter(([, v]) => v !== undefined)
+        Object.entries(obj)
+            .filter(([, v]) => v !== valueToRemove)
             .map(([k, v]) => [k, cast(v)])
     );
 }
 
+function sanitizeSearch(search) {
+    return sanitize(search);
```

Comment:
I'd explicitly set `undefined` for clarity, but it is only my very personal feeling

Reply:
https://github.com/odoo/odoo/pull/146070#discussion_r1448531831